### PR TITLE
Initial simd parsing

### PR DESCRIPTION
**Pull request by andreaTP**
Starting to get #528 in with reasonable chunks.

### CODE REVIEW

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java [hunk 1]
```diff
@@ -118,6 +118,15 @@ public boolean isReference() {
         }
     }
 
+    public boolean isVec() {
+        switch (this) {
+            case V128:
+                return true;
+            default:
+                return false;
+        }
```
Comment (electrum) [inline wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java:127]: Do we expect more types in the future? I only see `v128` in the specification.
```suggestion
        return this == V128;
```
Also, the method is currently unused. If there is only one vector type, seems better to just compare against the enum value.

File: wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java [hunk 2]
```diff
@@ -71,4 +71,57 @@ public void toStringContract() {
...
+        assertEquals(14, result[14]);
+        assertEquals(15, result[15]);
+    }
+
+    @Test
+    public void shouldConvertBackFromBytes() {
+        var value = new byte[] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
+        var result = Value.bytesToVec(value);
+        long xLow = 506097522914230528L;
+        long xHigh = 1084818905618843912L;
```
Comment (electrum) [inline wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java:121]: Much easier to read as hex:
```suggestion
        long xLow = 0x0706_0504_0302_0100L;
        long xHigh = 0x0F0E_0D0C_0B0A_0908L;
```

File: wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java [hunk 3]
```diff
@@ -71,4 +71,57 @@ public void toStringContract() {
...
+        assertEquals(4, result[4]);
+        assertEquals(5, result[5]);
+        assertEquals(6, result[6]);
+        assertEquals(7, result[7]);
+    }
+
+    @Test
+    public void shouldConvertToArraysHL() {
+        long xLow = 506097522914230528L;
+        long xHigh = 1084818905618843912L;
```
Comment (electrum) [inline wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java:94]:
```suggestion
        long xLow = 0x0706_0504_0302_0100L;
        long xHigh = 0x0F0E_0D0C_0B0A_0908L;
```

File: wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java [hunk 4]
```diff
@@ -71,4 +71,57 @@ public void toStringContract() {
         var i32FortyTwo = Value.i32(42);
         assertNotNull(i32FortyTwo.toString());
     }
+
+    @Test
+    public void shouldConvertToArrays() {
+        long x = 506097522914230528L;
```
Comment (electrum) [inline wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java:77]:
```suggestion
        long x = 0x0706_0504_0302_0100L;
```

**Review by electrum — APPROVED** [review]

File: wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java (hunk 2 above)
Comment (electrum) [reply wasm/src/test/java/com/dylibso/chicory/wasm/types/ValueTest.java:121]: If you name these `xLo` and `xHi`, the values will be aligned.

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java (hunk 1 above)
Comment (andreaTP) [reply wasm/src/main/java/com/dylibso/chicory/wasm/types/ValueType.java:127]: removed

File: wasm/src/test/java/com/dylibso/chicory/wasm/ParserTest.java [hunk 5]
```diff
@@ -221,4 +221,11 @@ public void shouldParseNamesSection() throws IOException {
             assertEquals(".rodata", nameSec.nameOfData(0));
         }
     }
+
+    @Test
+    public void shouldParseSIMD() throws IOException {
+        try (InputStream is = getClass().getResourceAsStream("/wasm/simd_load.0.wasm")) {
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply wasm/src/test/java/com/dylibso/chicory/wasm/ParserTest.java:227]: This is `spec.0.wasm` generated from `simd_load.wast`, I propose that we keep it in this PR and I'll remove as soon as we start running the wast generated tests.

**Pull request merged**